app/actions/websocket/roles.ts

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {fetchRolesIfNeeded, RolesRequest} from '@actions/remote/role';
+import {fetchMe} from '@actions/remote/user';
+import DatabaseManager from '@database/manager';
+import {queryRoleById} from '@queries/servers/role';
+import {queryCurrentUserId} from '@queries/servers/system';
+import {prepareMyTeams} from '@queries/servers/team';
+import {prepareUsers} from '@queries/servers/user';
+import {WebSocketMessage} from '@typings/api/websocket';
+
+import type {Model} from '@nozbe/watermelondb';
+import type {ServerDatabase} from '@typings/database/database';
+import type RoleModel from '@typings/database/models/servers/role';
+
+export async function handleRoleUpdatedEvent(serverUrl: string, msg: WebSocketMessage): Promise<void> {
+    const database = DatabaseManager.serverDatabases[serverUrl];
+    if (!database) {
+        return;
+    }
+
+    // only update Role records that exist in the Role Table
+    const role = JSON.parse(msg.data.role);
```

Comment:
JSON.parse may throw an exception, so cover it with a try/catch.

Reply:
I used the `safeParseJSON()` utility function from 

https://github.com/mattermost/mattermost-mobile/blob/gekidou/app/utils/helpers.ts#L89-L100

Reply:
```ts
export function safeParseJSON(rawJson: string | Record<string, unknown> | unknown[]) {
    let data = rawJson;
    try {
        if (typeof rawJson == 'string') {
            data = JSON.parse(rawJson);
        }
    } catch {
        // Do nothing
    }

    return data;
}
```

Reply:
Is this what we want thought? If `JSON.parse` fails we receive a string, instead of the type we are looking for.